scaaml/capture/scope/lecroy/lecroy_communication.py

```diff
@@ -173,27 +177,25 @@ def get_waveform(
     ) -> LecroyWaveform:  # pragma: no cover
         """Get a LecroyWaveform object representing a single waveform.
         """
-        assert self._scope
-
-        return self._scope.query_binary_values(
-            f"{channel}:WAVEFORM?",
-            datatype="B",
-            container=LecroyWaveform,
-        )  # type: ignore
+        return self.query_binary_values(f"{channel}:WAVEFORM?",
+                                        container=LecroyWaveform,
+                                        datatype="B")
 
     @make_custom_exception
-    def query_binary_values(self,
-                            message: str,
-                            datatype: BINARY_DATATYPES = "B",
-                            container=bytearray):  # pragma: no cover
+    def query_binary_values(
+            self,
+            message: str,
+            container: type[T],
+            datatype: BINARY_DATATYPES = "B") -> T:  # pragma: no cover
         """Query binary data."""
         assert self._scope
         self._logger.debug("query_binary_values(message=\"%s\")", message)
-        return self._scope.query_binary_values(
+        raw_data = self._scope.query_binary_values(
             message,
             datatype=datatype,
             container=container,
         )
+        return cast(T, raw_data)
```

Comment:
The PyVISA type is somehow broken. PyVISA type hints says that it returns `Sequence[Union[int, float]]` no matter the value of `container` (which is typed as `Union[Type, Callable[[Iterable], Sequence]]`).
And because `container` can be `Type`, it then can be anything, which will be called to build and return the results. That's what we're using to parse/build a `LeCroyWaveform`. 
So here I'm fixing the type hints by saying that `container` is a type and the function will return an instance of that type.
If we want to restrict the possible types, it can be done by specifying them at declaration of the `TypeVar`